guides/source/active_record_migrations.md

```diff
@@ -220,88 +243,89 @@ class RemovePartNumberFromProducts < ActiveRecord::Migration[8.0]
 end
 ```
 
-### Creating New Tables
+### Creating Associations
+
+Active Record associations are used to define relationships between different
+models in your application, allowing them to interact with each other through
+their relationships and making it easier to work with related data. To learn
+more about associations, you can refer to the [Association Basics
+guide](association_basics.html).
+
+One common use case for associations is creating foreign key references between
+tables. The generator accepts column types such as `references` to facilitate
+this process. [References](#references) are a shorthand for creating columns,
+indexes, foreign keys, or even polymorphic association columns.
 
-If the migration name is of the form "CreateXXX" and is followed by a list of
-column names and types then a migration creating the table XXX with the
-columns listed will be generated. For example:
+For example,
 
 ```bash
-$ bin/rails generate migration CreateProducts name:string part_number:string
+$ bin/rails generate migration AddUserRefToProducts user:references
 ```
 
-generates
+generates the following [`add_reference`][] call:
 
 ```ruby
-class CreateProducts < ActiveRecord::Migration[8.0]
+class AddUserRefToProducts < ActiveRecord::Migration[8.0]
   def change
-    create_table :products do |t|
-      t.string :name
-      t.string :part_number
-
-      t.timestamps
-    end
+    add_reference :products, :user, null: false, foreign_key: true
   end
 end
 ```
 
-As always, what has been generated for you is just a starting point.
-You can add or remove from it as you see fit by editing the
-`db/migrate/YYYYMMDDHHMMSS_create_products.rb` file.
-
-### Creating associations using references
-
-Also, the generator accepts column type as `references` (also available as
-`belongs_to`). For example,
+The above migration creates a foreign key called `user_id`in the `products`
+table, where `user_id` is a reference to the `id` column in the `users` table.
+It also creates an index for the `user_id` column. The schema looks as follows:
 
-```bash
-$ bin/rails generate migration AddUserRefToProducts user:references
+```ruby
+  t.uuid "user_id", null: false
```

Comment:
I think the `type` should be `bigint`

```suggestion
  t.bigint "user_id", null: false
```

Reply:
@ahmad19 In code review, it's important to state your reasons for taking a stance that contradicts what's in the PR.

Reply:
Yes. It should be `bigint` here, unless someone changes the default type of foreign keys. we should also show what is `t` here since there is no table reference in the code snippet. So maybe we should show the entire `create_table` code equivalent to the schema of the products table.